acceptance/tests/test_upgrade_pc1_to_puppet5.rb

```diff
@@ -2,29 +2,11 @@
 require_relative '../helpers'
 
 # Tests FOSS upgrades from the latest puppet 4 (the PC1 collection) to puppet 5.5.10.
-# This test will only run if the version of puppet on the master host is less than 5.
 test_name 'puppet_agent class: collection parameter for FOSS upgrades' do
-  master_agent_version = puppet_agent_version_on(master)
+  master_collection restrict_to: 'puppet5'
```

Comment:
I think this captures the general pattern of an agent upgrade test based on what I saw from the existing code.

The DSL helper `master_collection` replaces the existing `agent_version` checks. Reason I chose collection is b/c it is more consistent with how we think about upgrades.

`is_foss_upgrade_test` replaces the existing `PE_ONLY_UPGRADES` confine. Once we start testing PE upgrades here, we may likely have a similar `is_pe_upgrade_test` helper.

`setup_agents_to_upgrade` is essentially `prepare_upgrade_with` but there are some caveats:
* The method takes in an optional block for additional set-up. For example for a Windows upgrade test, you can do something like
```
setup_agents_to_upgrade('pc1') do
  <start services here>
end
```

* Any subsequent teardown blocks _after_ the invocation of `setup_agents_to_upgrade` _cannot_ use puppet, facter, i.e. anything from the puppet_agent package. For example, something like
```
setup_agents_to_upgrade('pc1')
teardown { on(agent, puppet('--version')) }
```

will fail because `setup_agents_to_upgrade` has some teardown blocks to clean the agents so that by the time we get to the declared teardown block, puppet-agent will have already been uninstalled. However, something like
```
setup_agents_to_upgrade('pc1')
on(agent, puppet('--version'))
```

will work.

There's a technical reason why `purge_agents`' code was moved over to `setup_agents_to_upgrade` that I'll explain in a separate PR comment underneath the method.


There's a technical reason why the block pattern that I'll discuss in a separate PR comment underneath the method.